src/document.jl

```diff
@@ -91,7 +91,7 @@ function parse_file(filename::String)
 end
 
 function parse_string(s::String)
-    p = ccall(xmlParseMemory, Xptr, (Ptr{Cchar}, Cint), s, sizeof(s) + 1)
+    p = ccall(xmlParseMemory, Xptr, (Ptr{Cchar}, Cint), pointer(s), sizeof(s) + 1)
```

Comment:
if you want argument `s` to be any `String` type then you need to call `bytestring(s)` here.